cupy/math/misc.py

```diff
@@ -1,8 +1,169 @@
+import scipy
+
+import cupy
+import cupyx
+
 from cupy import core
 from cupy.core import _routines_math as _math
 from cupy.core import fusion
+from cupyx.scipy import fft
+
+dot_kernel = core.ReductionKernel(
+    'T x1, T x2',
+    'T y',
+    'x1 * x2',
+    'a + b',
+    'y = a',
+    '0',
+    'dot_product'
+)
 
-# TODO(okuta): Implement convolve
+
+def convolve(a, v, mode='full'):
+    """Returns the discrete, linear convolution of two one-dimensional sequences.
+
+    Args:
+        a (cupy.ndarray): first 1-dimensional input.
+        v (cupy.ndarray): second 1-dimensional input.
+        mode (str, optional): `valid`, `same`, `full`
+
+    Returns:
+        cupy.ndarray: Discrete, linear convolution of a and v.
+
+    .. seealso:: :func:`numpy.convolve`
+
+    """
+    if a.ndim == 0:
+        a = a.ravel()
+    if v.ndim == 0:
+        v = v.ravel()
+    if v.size > a.size:
+        a, v = v, a
+    if a.size == 0:
+        raise ValueError('a cannot be empty')
+    if v.size == 0:
+        raise ValueError('v cannot be empty')
+    if v.ndim > 1:
+        raise ValueError('v cannot be multidimensional array')
+
+    method = cupyx.scipy.signal.choose_conv_method(a, v, mode)
+    if method == 'direct':
+        _, out = _dot_convolve(a, v[::-1], mode)
+    elif method == 'fft':
+        out = _fftconvolve1d(a, v, mode)
+    else:
+        raise ValueError('Unsupported method')
+    return out
+
+
+def _fftconvolve1d(a1, a2, mode='full'):
+
+    inverted = 0
+    if a2.size > a1.size:
+        a1, a2 = a2, a1
+        inverted = not a2.size % 2
```

Comment:
For the readability.
```suggestion
        inverted = a2.size % 2 == 1
```